src/components/Popups/PopupDrawingToolbar.tsx

```diff
@@ -57,9 +59,25 @@ const PopupDrawingToolbar = ({
     reference,
 }: Props): JSX.Element => {
     const intl = useIntl();
+    const popupRef = React.useRef<PopupBase>(null);
+    const prevReference = usePrevious(reference.getBoundingClientRect());
```

Comment:
Could `reference.getBoundingClientRect()` be shared between this line and in the `useEffect` callback?